public/app/features/correlations/types.ts

```diff
@@ -57,9 +57,15 @@ export type CorrelationType = 'query' | 'external'; //(typeof corrTypeArray)[num
 
 export type ExternalTypeTarget = { url: string };
 
-export type CorrelationConfig = {
+export type CorrelationConfigQuery = {
```

Comment:
So the config now also can be separate so there is less chance of having different configs for different correlation type.